jenkins/opensearch-dashboards/bwc-test.jenkinsfile

```diff
@@ -20,20 +22,40 @@ pipeline {
             description: 'The build manifest URL, e.g. https://ci.opensearch.org/ci/dbc/distribution-build-opensearch-dashboards/2.0.0/98/linux/x64/builds/opensearch-dashboards/manifest.yml.',
             trim: true
         )
+        string(
+            name: 'BUILD_BRANCH',
+            description: "Build repo branch to use.",
+            defaultValue: "main",
+            trim: true
+        )
```

Comment:
This feels very wrong. This job itself lives in this repo in a branch. If I am running the job out branch X, I would not expect the job to check out build tools from branch Y.

Reply:
This is done for the fact Jenkinsfiles cannot be branched, they has to reside in `main` and jenkins job git SCM configuration uses single branch to pull the jenkinsfile, once jenkinsfile is pulled, depending on the branch passed as input using `BUILD_BRANCH`, then the checkout happens with right branch and all the stages are executed as per the flow.